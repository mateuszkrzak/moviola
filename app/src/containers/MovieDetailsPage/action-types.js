export const FETCH_MOVIE_BY_ID_REQUEST = 'FETCH_MOVIE_BY_ID_REQUEST';
export const FETCH_MOVIE_BY_ID_FAILURE = 'FETCH_MOVIE_BY_ID_FAILURE';
export const FETCH_MOVIE_BY_ID_SUCCESS = 'FETCH_MOVIE_BY_ID_SUCCESS';
export const FETCH_MOVIES_BY_GENRES_REQUEST = 'FETCH_MOVIES_BY_GENRES_REQUEST';
export const FETCH_MOVIES_BY_GENRES_FAILURE = 'FETCH_MOVIES_BY_GENRES_FAILURE';
export const FETCH_MOVIES_BY_GENRES_SUCCESS = 'FETCH_MOVIES_BY_GENRES_SUCCESS';
