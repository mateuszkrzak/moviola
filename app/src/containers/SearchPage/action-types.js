export const FETCH_MOVIES_REQUEST = 'FETCH_MOVIES_REQUEST';
export const FETCH_MOVIES_FAILURE = 'FETCH_MOVIES_FAILURE';
export const FETCH_MOVIES_SUCCESS = 'FETCH_MOVIES_SUCCESS';
export const SET_MOVIES_QUERY = 'SET_MOVIES_QUERY';
export const SET_MOVIES_SORT_BY = 'SET_MOVIES_SORT_BY';
export const SET_MOVIES_SEARCH_BY = 'SET_MOVIES_SEARCH_BY';