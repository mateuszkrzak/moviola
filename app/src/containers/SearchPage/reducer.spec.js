import { FETCH_MOVIES_REQUEST, FETCH_MOVIES_FAILURE, FETCH_MOVIES_SUCCESS } from './action-types';
import searchReducer from './reducer';

describe('Search reducer', () => {
  const initialState = {
    movies: [],
    error: '',
    isLoading: false,
    moviesCount: 0,
  };

  const mockMovie = {
    title: 'Test tile',
    release_date: '2020-10-20',
    vote_average: 9,
    poster_path: 'http://poster.path',
    runtime: 120,
    overview: 'Test overview',
    genres: ['Test genre', 'Another test genre'],
  };

  it('should return the initial state', () => {
    const action = {
      type: null,
    };

    const testedReducer = searchReducer(undefined, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should set isLoading after FETCH_MOVIES_REQUEST action', () => {
    const action = {
      type: FETCH_MOVIES_REQUEST,
    };

    const testedReducer = searchReducer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should set movies after FETCH_MOVIES_SUCCESS action', () => {
    const action = {
      payload: { movies: [mockMovie, mockMovie, mockMovie], moviesCount: 3 },
      type: FETCH_MOVIES_SUCCESS,
    };

    const testedReducer = searchReducer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });

  it('should handle FETCH_MOVIES_FAILURE action', () => {
    const action = {
      type: FETCH_MOVIES_FAILURE,
      payload: 'Test error message',
    };

    const testedReducer = searchReducer(initialState, action);

    expect(testedReducer).toMatchSnapshot();
  });
});
