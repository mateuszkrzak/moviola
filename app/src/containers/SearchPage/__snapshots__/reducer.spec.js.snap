// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search reducer should handle FETCH_MOVIES_FAILURE action 1`] = `
Object {
  "error": "Test error message",
  "isLoading": false,
  "movies": Array [],
  "moviesCount": 0,
}
`;

exports[`Search reducer should return the initial state 1`] = `
Object {
  "error": "",
  "isLoading": false,
  "movies": Array [],
  "moviesCount": 0,
}
`;

exports[`Search reducer should set isLoading after FETCH_MOVIES_REQUEST action 1`] = `
Object {
  "error": "",
  "isLoading": true,
  "movies": Array [],
  "moviesCount": 0,
}
`;

exports[`Search reducer should set movies after FETCH_MOVIES_SUCCESS action 1`] = `
Object {
  "error": "",
  "isLoading": false,
  "movies": Array [
    Object {
      "genres": Array [
        "Test genre",
        "Another test genre",
      ],
      "overview": "Test overview",
      "poster_path": "http://poster.path",
      "release_date": "2020-10-20",
      "runtime": 120,
      "title": "Test tile",
      "vote_average": 9,
    },
    Object {
      "genres": Array [
        "Test genre",
        "Another test genre",
      ],
      "overview": "Test overview",
      "poster_path": "http://poster.path",
      "release_date": "2020-10-20",
      "runtime": 120,
      "title": "Test tile",
      "vote_average": 9,
    },
    Object {
      "genres": Array [
        "Test genre",
        "Another test genre",
      ],
      "overview": "Test overview",
      "poster_path": "http://poster.path",
      "release_date": "2020-10-20",
      "runtime": 120,
      "title": "Test tile",
      "vote_average": 9,
    },
  ],
  "moviesCount": 3,
}
`;
